message.go

```diff
@@ -43,6 +43,14 @@ func IsValkeyBusyGroup(err error) bool {
 	return false
 }
 
+// IsValkeyRedirect checks if it is a valkey REDIRECT message.
+func IsValkeyRedirect(err error) (addr string, ok bool) {
```

Comment:
Can we try to remove this function? I think a `if ret, yes := IsValkeyErr(err); yes && ret. IsRedirect {` is enough.